### PR TITLE
Fixed request parameters on PostController

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

File: Controller/PostController.php [hunk 1]
```diff
@@ -23,9 +23,11 @@
 class PostController extends Controller
 {
     /**
+     * @param Request $request
+     *
      * @return RedirectResponse
      */
-    public function homeAction()
+    public function homeAction(Request $request = null)
```
Comment (Soullivaneuh) [inline Controller/PostController.php]: This is not necessary. The request is never used here.

Comment (Soullivaneuh) [inline Controller/PostController.php]: Same for other unused requests.

File: Controller/PostController.php [hunk 2]
```diff
@@ -165,8 +167,6 @@ public function archiveYearlyAction($year, Request $request = null)
      */
     public function viewAction($permalink, Request $request = null)
     {
-        $request = $this->resolveRequest($request);
```
Comment (Soullivaneuh) [inline Controller/PostController.php]: Why removing this?

File: Controller/PostController.php (hunk 1 above)
Comment (core23) [inline Controller/PostController.php]: Yes, that's true, but when extending this controller you could easily access the request attribute if needed.

File: Controller/PostController.php (hunk 2 above)
Comment (core23) [inline Controller/PostController.php]: It was a mistake in my previous PR. The request is never used in the viewAction.

Comment (Soullivaneuh) [inline Controller/PostController.php]: In this case, you can remove it from method signature.

File: Controller/PostController.php (hunk 1 above)
Comment (Soullivaneuh) [inline Controller/PostController.php]: I don't think so. If you override the method, you can access the request in any case.

This parameter is always added AFAIK.

Comment (core23) [inline Controller/PostController.php]: I tried adding the request parameter to the homeAction in my own PostController:
```php

    public function homeAction(Request $request)
    {
        return $this->redirect($this->generateUrl('sonata_news_archive'));
    }
```
Got an exception:
```
Runtime Notice: Declaration of Core23\NewsBundle\Controller\PostController::homeAction() 
should be compatible with Sonata\NewsBundle\Controller\PostController::homeAction()
```

Comment (Soullivaneuh) [inline Controller/PostController.php]: Should work with `Request $request = null`.

But, right, this could be indeed be simplier.

Comment (core23) [inline Controller/PostController.php]: Forget about my last comment. 

You could use

```php
    public function homeAction(Request $request=null)
```

**Comment by OskarStark** [comment]
IMO we added the resolve method a few weeks ago to stay BC 

@core23 please check this out. Thank you

**Comment by core23** [comment]
Yes, you're right. But I missed out that the request in the ``viewAction`` is never used.

File: Controller/PostController.php (hunk 1 above)
Comment (Soullivaneuh) [inline Controller/PostController.php]: Yes, but you're right. If your override don't provide parameter, this will not work.

**Comment by OskarStark** [comment]
Thank you @core23

**Pull request merged**